# HG changeset patch
# User Lukas Reichlin <lukas.reichlin@gmail.com>
# Date 1324264550 28800
#      Sun Dec 18 19:15:50 2011 -0800
# Branch stable
# Node ID b2e9c4b0c4f8ca671db1cf687e021eaa27d46b48
# Parent  80bda7c4f01232795db4928d312600448d201d0c
isdefinite.m: Use any specified tolerance in deciding whether matrix is hermitian.

* isdefinite.m: Use any specified tolerance in deciding whether matrix is hermitian.

diff --git a/scripts/linear-algebra/isdefinite.m b/scripts/linear-algebra/isdefinite.m
--- a/scripts/linear-algebra/isdefinite.m
+++ b/scripts/linear-algebra/isdefinite.m
@@ -32,26 +32,26 @@
 ## Adapted-By: jwe
 
 function retval = isdefinite (x, tol)
 
   if (nargin < 1 || nargin > 2)
     print_usage ();
   endif
 
-  if (! ishermitian (x))
-    error ("isdefinite: X must be a hermitian matrix");
-  endif
-
   if (! isfloat (x))
     x = double (x);
   endif
 
   if (nargin == 1)
-    tol = 100 * eps(class (x)) * norm (x, "fro");
+    tol = 100 * eps (class (x)) * norm (x, "fro");
+  endif
+
+  if (! ishermitian (x, tol))
+    error ("isdefinite: X must be a Hermitian matrix");
   endif
 
   e = tol * eye (rows (x));
   [r, p] = chol (x - e);
   if (p == 0)
     retval = 1;
   else
     [r, p] = chol (x + e);
@@ -59,16 +59,17 @@ function retval = isdefinite (x, tol)
       retval = 0;
     else
       retval = -1;
     endif
   endif
 
 endfunction
 
+
 %!test
 %! A = [-1 0; 0 -1];
 %! assert (isdefinite (A), -1)
 
 %!test
 %! A = [1 0; 0 1];
 %! assert (isdefinite (A), 1)
 
@@ -76,9 +77,11 @@ endfunction
 %! A = [2 -1 0; -1 2 -1; 0 -1 2];
 %! assert (isdefinite (A), 1)
 
 %!test
 %! A = [1 0; 0 0];
 %! assert (isdefinite (A), 0)
 
 %!error isdefinite ()
-%!error isdefinite ([1 2; 3 4])
\ No newline at end of file
+%!error isdefinite (1,2,3)
+%!error <X must be a Hermitian matrix> isdefinite ([1 2; 3 4])
+
